client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -39,15 +39,19 @@ const doAppPushRequest = ( store ) => {
 		.accept( 'application/json' )
 		.send( {
 			user_id: getTwoFactorUserId( store.getState() ),
-			two_step_nonce: getTwoFactorAuthNonce( store.getState() ),
+			two_step_nonce: getTwoFactorAuthNonce( store.getState(), 'push' ),
 			remember_me: getTwoFactorRememberMe( store.getState() ),
 			two_step_push_token: getTwoFactorPushToken( store.getState() ),
 			client_id: config( 'wpcom_signup_id' ),
 			client_secret: config( 'wpcom_signup_key' ),
 		} ).then( () => {
 			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED } );
 		} ).catch( error => {
-			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_UPDATE_NONCE, twoStepNonce: error.response.body.data.two_step_nonce } );
+			store.dispatch( {
```

Comment:
`TWO_FACTOR_AUTHENTICATION_UPDATE_NONCE` on the list of actions. Are you sure it comes from a different network request? :)